File: examples/python/cognito_ses_domain_example/example_stack.py
from constructs import Construct
import aws_cdk as cdk
from aws_cdk import (
    aws_cognito as cognito,
    aws_ses as ses,
)

# After building the library with `npx projen package`, install python dist:
#   pip install ../../dist/python/cognito_ses_domain-<version>.tar.gz
# Then import the construct below.
from cognito_ses_domain import SesDomainIdentity
from ses_cloudwatch import SesCloudWatchProps

class ExampleStack(cdk.Stack):
    def __init__(self, scope: Construct, construct_id: str, **kwargs) -> None:
        super().__init__(scope, construct_id, **kwargs)

        user_pool = cognito.UserPool(self, "UserPool")
        client = user_pool.add_client("UserPoolClient")

        SesDomainIdentity(
            self,
            "SesDomainIdentityExample",
            domain="playground.amer.merapar.net",  # must exist in your Route53 hosted zones when deploying
            user_pool=user_pool,
            user_pool_client_id=client.user_pool_client_id,
            # create_identity_pool=True,  # Uncomment to provision Identity Pool & role
            sending_logs=SesCloudWatchProps(
                configuration_set_name="python-example-conf-set",
                log_group_name="python-example-log-group",
                event_rule_name="python-example-event-rule",
                events=[ses.EmailSendingEvent.SEND, ses.EmailSendingEvent.DELIVERY],
            )
        )

File: examples/typescript/lib/example-stack.ts
import { Stack, StackProps, RemovalPolicy } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import * as cognito from 'aws-cdk-lib/aws-cognito';
import { SesDomainIdentity } from 'cognito-ses-domain';
import * as ses from 'aws-cdk-lib/aws-ses';

export class ExampleStack extends Stack {
  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);

    let domain = 'playground.amer.merapar.net'; // must exist in your Route53 hosted zones when deploying

    // User pool and client
    const userPool = new cognito.UserPool(this, 'UserPool', {
      selfSignUpEnabled: true,
      removalPolicy: RemovalPolicy.DESTROY,
      email: cognito.UserPoolEmail.withSES({
        fromEmail: 'noreply@playground.amer.merapar.net',
        fromName: 'Playground',
        sesVerifiedDomain: domain,
      }),
    });
    const client = userPool.addClient('UserPoolClient', { generateSecret: false });

    // Demonstrate SES domain identity with sending logs and no identity pool by default
    new SesDomainIdentity(this, 'SesDomainIdentityExample', {
      domain,
      userPool,
      userPoolClientId: client.userPoolClientId,
      // createIdentityPool: true, // uncomment to provision Identity Pool + role
      sendingLogs: {
        configurationSetName: 'example-ses-config-set',
        events: [ses.EmailSendingEvent.SEND, ses.EmailSendingEvent.DELIVERY],
      },
    });
  }
}
